x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/filter/SecurityActionFilter.java

```diff
@@ -150,9 +150,10 @@ the system itself (e.g. pings, update mappings, share relocation, etc...) and we
          it to the action without an associated user (not via REST or transport - this is taken care of by
          the {@link Rest} filter and the {@link ServerTransport} filter respectively), it's safe to assume a system user
          here if a request is not associated with any other user.
+         Because we want to fallback to the SystemUser, we don't allow anonymous (AnonymousUser) requests.
```

Comment:
fallback user is checked before the anonymous user so we would never end up with `AnonymousUser` even if we passed `true` here. Not sure if worth mentioning in the javadoc, I was just confused while reading the existing one. Maybe something like

> We want to fallback to the SystemUser and this takes precedence, but we explicitly disallow AnonymousUser fallback in order to be future proof

I don't like that much either tbh

Reply:
I've split this method so you can either pass in 1, non-null fallback user, or a boolean to control anonymous access. You can no longer pass both.